src/three-domain-secure/component.jsx

```diff
@@ -115,47 +60,75 @@ const parseMerchantPayload = ({
 };
 
 export interface ThreeDomainSecureComponentInterface {
-  isEligible(): Promise<boolean>;
-  show(): ZoidComponent<void>;
+  isEligible(payload: MerchantPayloadData): Promise<boolean>;
+  show(): ZalgoPromise<threeDSResponse>;
 }
+
+type Update3DSTokenResponse = {|
+  updateTokenizedCreditCardWithExternalThreeDSecure: {|
+    paymentMethod: {|
+      id: string,
+    |},
+  |},
+|};
+
 export class ThreeDomainSecureComponent {
+  fastlaneNonce: string;
   logger: LoggerType;
-  request: Request;
+  restClient: RestClient;
+  graphQLClient: GraphQLClient;
   sdkConfig: SdkConfig;
   authenticationURL: string;
+  threeDSIframe: ZoidComponent<TDSProps>;
 
   constructor({
     logger,
-    request,
+    restClient,
+    graphQLClient,
     sdkConfig,
   }: {|
     logger: LoggerType,
-    request: Request,
+    restClient: RestClient,
+    graphQLClient: GraphQLClient,
     sdkConfig: SdkConfig,
   |}) {
     this.logger = logger;
-    this.request = request;
+    this.restClient = restClient;
+    this.graphQLClient = graphQLClient;
     this.sdkConfig = parseSdkConfig({ sdkConfig, logger });
   }
 
   async isEligible(merchantPayload: MerchantPayloadData): Promise<boolean> {
     const data = parseMerchantPayload({ merchantPayload });
+    this.fastlaneNonce = merchantPayload.nonce;
+
+    try {
+      const accessToken = await createAccessToken(this.sdkConfig.clientID);
+      // $FlowIssue confusing ZalgoPromise return type with resolved string value
+      this.restClient.setAccessToken(accessToken);
+    } catch (error) {
+      this.logger.warn(error);
+      throw error;
+    }
+
     try {
       // $FlowFixMe
-      const { status, links } = await this.request<requestData, responseBody>({
+      const { status, links } = await this.restClient.request<
+        requestData,
+        responseBody
+      >({
         method: "POST",
-        url: `${this.sdkConfig.paypalApiDomain}/${PAYMENT_3DS_VERIFICATION}`,
```

Comment:
Are we OK with keeping this a constant from `src/constant/api.js`? It wasn't clear why this was changed.